src/ripple/app/misc/impl/Transaction.cpp

```diff
@@ -132,6 +133,23 @@ Transaction::pointer Transaction::load(uint256 const& id, Application& app)
         soci::blob sociRawTxnBlob (*db);
         soci::indicator rti;
 
+        if(searchedAll)
+        {
+            uint64_t count = 0;
+
+            *db << "SELECT COUNT(DISTINCT LedgerSeq) FROM Transactions WHERE LedgerSeq BETWEEN "
+                << range.first ()
+                << " AND "
+                << range.last ()
```

Comment:
I'm unsure how expensive this function is when the range is large. I think we want to restrict the range to a reasonable size. @intelliot has suggested a range of 200 would be sufficient. I'm fine making it somewhat bigger than that if we want. I'll suggest 500 or 1000.

Reply:
Should we return an error if the range exceeds the limit?

Reply:
Yes, I would return an error.

Reply:
Done